pymc3/distributions/bound.py

```diff
@@ -75,40 +81,64 @@ class Bound(Distribution):
                     'par2', mu=0.0, sigma=1.0, initval=1.0)
     """
 
+    rv_op = boundrv
+
+    def __new__(cls, *args, **kwargs):
+        transform = kwargs.get("transform", UNSET)
+        if transform is UNSET:
+            kwargs["transform"] = cls.default_transform(kwargs)
+        return super().__new__(cls, *args, **kwargs)
+
     @classmethod
-    def dist(self, distribution, lower=None, upper=None):
+    def default_transform(cls, kwargs_dict):
```

Comment:
This might work out of the box if you inherit from `BoundedContinuous`, which applies this logic by default. You just need to tell what are the indexes of the arguments that specify "lower" and "upper", which in this case is (1, 2). There are some examples in `distributions/continuous.py`. In either case you still need to override transform to be `None` for discrete variables.